src/goose/cli/prompt/goose_prompt_session.py

```diff
@@ -1,34 +1,64 @@
 from typing import Optional
 
 from prompt_toolkit import PromptSession
+from prompt_toolkit.document import Document
 from prompt_toolkit.formatted_text import FormattedText
 from prompt_toolkit.validation import DummyValidator
 
 from goose.cli.prompt.create import create_prompt
+from goose.cli.prompt.lexer import PromptLexer
 from goose.cli.prompt.prompt_validator import PromptValidator
 from goose.cli.prompt.user_input import PromptAction, UserInput
+from goose.command import get_commands
 
 
 class GoosePromptSession:
-    def __init__(self, prompt_session: PromptSession) -> None:
-        self.prompt_session = prompt_session
+    def __init__(self) -> None:
+        # instantiate the commands available in the prompt
+        self.commands = dict()
+        command_plugins = get_commands()
+        for command, command_cls in command_plugins.items():
+            self.commands[command] = command_cls()
+        self.main_prompt_session = create_prompt(self.commands)
+        self.text_prompt_session = PromptSession()
 
-    @staticmethod
-    def create_prompt_session() -> "GoosePromptSession":
-        return GoosePromptSession(create_prompt())
+    def get_message_after_commands(self, message: str) -> str:
```

Comment:
I added some comments to make the flow of logic clearer to the uninitiated but feel free to ignore!